ark/analysis/spatial_analysis.py

```diff
@@ -1,12 +1,77 @@
 import pandas as pd
 import xarray as xr
 import numpy as np
-from ark.utils import spatial_analysis_utils
-from ark.utils import misc_utils
+from ark.utils import spatial_analysis_utils, misc_utils, load_utils, io_utils
 
 import ark.settings as settings
 
 
+def batch_channel_spatial_enrichment(label_dir, marker_thresholds, all_data, batch_size=5,
+                                     **kwargs):
+    """Wrapper function for batching calls to `calculate_channel_spatial_enrichment` over fovs
+
+    Args:
+        label_dir (str | Pathlike):
+            directory containing labeled tiffs
+        marker_thresholds (numpy.ndarray):
+            threshold values for positive marker expression
+        all_data (pandas.DataFrame):
+            data including fovs, cell labels, and cell expression matrix for all markers
+        batch_size (int):
+            fov count to load into memory at a time
+        **kwargs (dict):
+            args passed to `calculate_channel_spatial_enrichment`
+
+    Returns:
+        tuple (list, xarray.DataArray):
+
+        - a list with each element consisting of a tuple of closenum and closenumrand for each
+          fov included in the analysis
+        - an xarray with dimensions (fovs, stats, num_channels, num_channels). The included
+          stats variables for each fov are z, muhat, sigmahat, p, h, adj_p, and
+          cluster_names
+    """
+
+    # parse files in label_dir
+    all_label_names = io_utils.list_files(label_dir, substrs=['.tif'])
+
+    batching_strategy = \
+        [all_label_names[i:i + batch_size] for i in range(0, len(all_label_names), batch_size)]
+
+    # create containers for batched return values
+    values = []
+    stats_datasets = []
+
+    # extract 'included_fovs' kwarg
+    included_fovs = kwargs.get('included_fovs', None)
```

Comment:
We should filter `all_label_names` instead, right? Then we don't have to compute/batch over FOVs we don't care about?